src/components/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import { textKilo } from '../../styles/style-helpers';
+
+const baseStyles = ({ theme }) => css`
+  label: tooltip;
+  position: relative;
+  display: inline-block;
+
+  &:hover {
+    > span {
+      visibility: visible;
+      opacity: 1;
+    }
+  }
+`;
+
+const arrowAligns = theme => ({
+  center: `left: 50%`,
+  right: `left: ${theme.spacings.mega}`,
+  left: `right: ${theme.spacings.mega}`
+});
+
+const tooltipAlign = {
+  center: `
+    left: 0;
+    right: 0;
+    margin-left: auto;
+    margin-right: auto;
+  `,
+  right: 'left: 50%;',
+  left: 'right: 50%;'
+};
+
+const tooltipTextStyles = ({ align, theme }) => css`
+  label: tooltip_text;
+  visibility: hidden;
+  opacity: 0;
+  width: max-content;
+  min-width: 60px;
+  background-color: ${theme.colors.black};
+  color: ${theme.colors.white};
+  text-align: center;
+  border-radius: ${theme.borderRadius.mega};
+  padding: ${theme.spacings.byte};
+  position: absolute;
+  z-index: 1;
+  bottom: 125%;
+  ${tooltipAlign[align]};
+  transition: opacity 0.3s;
+  ${textKilo({ theme })};
+
+  &:after {
+    content: '';
+    position: absolute;
+    margin-left: -${theme.spacings.bit};
+    border-width: ${theme.spacings.bit};
+    border-style: solid;
+    border-color: ${theme.colors.black} transparent transparent transparent;
+    top: 100%;
+    ${arrowAligns(theme)[align]};
+  }
+`;
+
+const TooltipText = styled('span')(tooltipTextStyles);
+const TooltipElement = styled('div')(baseStyles);
+
+/**
+ * A Tooltip wrapper in order to show tooltips ontop of other
+ * components.
+ */
+const Tooltip = ({ children, content, align, ...props }) => {
+  return (
+    <TooltipElement {...props}>
+      <TooltipText {...{ align }}>{content}</TooltipText>
+      {children}
+    </TooltipElement>
+  );
+};
+
+Tooltip.Center = 'center';
+Tooltip.Left = 'left';
+Tooltip.Right = 'right';
+
+Tooltip.propTypes = {
+  /**
+   * The content inside of the tooltip being show.
+   */
+  content: PropTypes.node.isRequired,
```

Comment:
Could these be node or string?

Reply:
Actually, I didn't want to fix a type of this. On examples I we only have `text` but isn't some icon also acceptable too?